File: 1200-minimum-absolute-difference/main.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        arr.sort()
        _min = float('inf')

        n = len(arr) -1
        for i in range(n):
            _min = min(_min, abs(arr[i] - arr[i+1]))
        
        ans = []
        for i in range(n):
            if abs(arr[i] - arr[i+1]) == _min:
                ans.append([arr[i], arr[i+1]])

        return ans

File: 1200-minimum-absolute-difference/main.go
func minimumAbsDifference(arr []int) [][]int {
    sort.Slice(arr, func(i, j int) bool {
        return arr[i] < arr[j]
    })

    n := len(arr) - 1
    _min := int(math.MaxInt64)
    for i := range arr[:n] {
        _min = min(_min, abs(arr[i] - arr[i+1]))
    }

    var ans [][]int
    for i := range arr[:n]{
        if abs(arr[i] - arr[i+1]) == _min {
            ans = append(ans, []int{arr[i], arr[i+1]})
        }
    }

    return ans

}

func abs(a int) int {
    if a < 0 {
        return -a
    }
    return a
}

func min(a, b int) int {
    if a > b {
        return b
    }
    return a
}
